Source/WebCore/rendering/RenderLayer.h

```diff
@@ -770,7 +774,7 @@ class RenderLayer final : public CanMakeSingleThreadWeakPtr<RenderLayer>, public
     
     LayoutRect repaintRectIncludingNonCompositingDescendants() const;
 
-    void setRepaintStatus(RepaintStatus status) { m_repaintStatus = status; }
+    void setRepaintStatus(RepaintStatus status) { m_repaintStatus = status; setNeedsPositionUpdate(); }
```

Comment:
I'd appreciate a comment to say why changing the repaint status implies we need to call `setNeedsPositionUpdate()`. Should you check that the status changed?